lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
+      .then(({ body }) => {
+        const apiResponse = {
+          _id: body._id,
+          _version: body._version
+        };
+
+        return apiResponse;
+      })
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Replace a document to ElasticSearch
    *
-   * @param {Request} request
-   * @returns {Promise} resolve an object that contains _id
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {string} id - Document id
+   * @param {object} content - Document content
+   * @param {object} options - refresh (null), userId (null)
+   *
+   * @returns {Promise} { _id, _version, _source }
    */
-  replace(request) {
+  replace (index, collection, id, content,  { refresh=null, userId=null }) {
     const
-      esRequest = initESRequest(request, ['refresh']),
-      existQuery = {
-        index: esRequest.index,
-        type: esRequest.type,
-        id: request.input.resource._id
-      },
-      userId = this._getUserId(request);
+      esIndex = this._getESIndex(index, collection),
+      esRequest = {
+        index: esIndex,
+        id,
+        body: content,
+        refresh
+      };
 
     esRequest.id = request.input.resource._id;
     esRequest.body = request.input.body;
 
     assertNoRouting(esRequest);
     assertWellFormedRefresh(esRequest);
 
-    return this.kuzzle.indexCache.exists(esRequest.index, esRequest.type)
+    return this.kuzzle.indexCache.exists(index, collection)
       .then(exists => {
         if (! exists) {
           this.throw(
             'index_or_collection_does_not_exist',
-            esRequest.index,
-            esRequest.type
-          );
+            index,
+            collection);
         }
 
         // Add metadata
         esRequest.body._kuzzle_info = {
-          author: userId,
+          author: String(userId),
           createdAt: Date.now(),
           updatedAt: Date.now(),
-          updater: userId,
-          active: true,
-          deletedAt: null
+          updater: String(userId)
         };
-        // extends the response with the source from request
-        // When we write in ES, the response doesn't contain the initial document content
-        return this.client.exists(existQuery);
+
+        return this.client.exists({ index: esIndex, id });
       })
-      .then(exist => {
-        if (exist) {
-          return this.client.index(esRequest);
+      .then(({ body: exists }) => {
+        if (! exists) {
+          return this.reject('document_not_found', id);
         }
 
-        return Bluebird.reject(this.getError(
-          'document_not_found',
-          esRequest.id
-        ));
+        return this.client.index(esRequest);
       })
-      .then(result => this.refreshIndexIfNeeded(
-        esRequest,
-        _.extend(result, {_source: request.input.body}))
-      );
+      .then(({ body }) => {
+        const apiResponse = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "apiResponse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)